builtin/providers/google/resource_google_project.go

```diff
@@ -95,45 +138,69 @@ func resourceGoogleProjectCreate(d *schema.ResourceData, meta interface{}) error
 
 		// Apply the merged policy
 		log.Printf("[DEBUG] Setting new policy for project: %#v", p)
-		_, err = config.clientResourceManager.Projects.SetIamPolicy(project,
+		_, err = config.clientResourceManager.Projects.SetIamPolicy(pid,
 			&cloudresourcemanager.SetIamPolicyRequest{Policy: p}).Do()
 
 		if err != nil {
-			return fmt.Errorf("Error applying IAM policy for project %q: %s", project, err)
+			return fmt.Errorf("Error applying IAM policy for project %q: %s", pid, err)
 		}
 	}
-	return nil
+
+	return resourceGoogleProjectRead(d, meta)
 }
 
 func resourceGoogleProjectRead(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	project, err := getProject(d, config)
-	if err != nil {
-		return err
-	}
-	d.SetId(project)
+	pid := d.Id()
 
-	// Confirm the project exists.
-	// TODO(evanbrown): Support project creation
-	p, err := config.clientResourceManager.Projects.Get(project).Do()
+	// Read the project
+	p, err := config.clientResourceManager.Projects.Get(pid).Do()
 	if err != nil {
 		if v, ok := err.(*googleapi.Error); ok && v.Code == http.StatusNotFound {
-			return fmt.Errorf("Project %q does not exist. The Google provider does not currently support new project creation.", project)
+			return fmt.Errorf("Project %q does not exist.", pid)
 		}
-		return fmt.Errorf("Error checking project %q: %s", project, err)
+		return fmt.Errorf("Error checking project %q: %s", pid, err)
 	}
 
+	d.Set("project_id", pid)
 	d.Set("number", strconv.FormatInt(int64(p.ProjectNumber), 10))
 	d.Set("name", p.Name)
 
+	if p.Parent != nil {
+		d.Set("org_id", p.Parent.Id)
+	}
+
+	// Read the IAM policy
+	pol, err := getProjectIamPolicy(pid, config)
+	if err != nil {
+		return err
+	}
+	d.Set("policy_etag", pol.Etag)
+
 	return nil
 }
 
 func resourceGoogleProjectUpdate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	project, err := getProject(d, config)
+	pid := d.Id()
+
+	// Read the project
+	p, err := config.clientResourceManager.Projects.Get(pid).Do()
 	if err != nil {
-		return err
+		if v, ok := err.(*googleapi.Error); ok && v.Code == http.StatusNotFound {
+			return fmt.Errorf("Project %q does not exist.", pid)
+		}
+		return fmt.Errorf("Error checking project %q: %s", pid, err)
+	}
+
+	// Project name has changed
+	if ok := d.HasChange("name"); ok {
+		p.Name = d.Get("name").(string)
+		// Do update on project
+		p, err = config.clientResourceManager.Projects.Update(p.ProjectId, p).Do()
+		if err != nil {
+			return fmt.Errorf("Error updating project %q: %s", p.Name, err)
+		}
 	}
 
 	// Policy has changed
```

Comment:
Is this not the same as `updateProjectIamPolicy`? It looks like it does the same thing. Maybe use that new abstraction here?